clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/provider/agent/KubernetesServerGroupCachingAgent.groovy

```diff
@@ -110,8 +114,17 @@ class KubernetesServerGroupCachingAgent extends KubernetesCachingAgent implement
       loadReplicaSet(namespace, serverGroupName)
     }
 
+    V1beta1DaemonSet daemonSet = metricsSupport.readData {
```

Comment:
It's starting to feel like these should be broken into separate caching agents. Having all of these done in a single caching agent increases the chance that the full cache cycle will be discarded on failure.